src/create_agent/check_agent.jl

```diff
@@ -0,0 +1,35 @@
+"""
+
+Function for checking the structure of the agent
+"""
+function check_agent(agent::Agent)
+
+   if length(agent.shared_parameters)> 0
+
+    ## Check for the same derived parameter in multiple shared parameters 
+    #Get out the derived parameters of all shared parameters
+    vector_of_derived_parameters = [agent.shared_parameters[i].derived_parameters for i in unique(keys(agent.shared_parameters))]
```

Comment:
Fjern unique, unødvendigt i dictionary hvor keys altid er unique.
Foreslår at sætte de to list comprehensions sammen i én.
kald det 'derived_parameters_vector' og så 'derived_parameters' måske.
Ik brug 'v' og 'y', men noget man kan læse hvad er, foreslår jeg. Det har været måden indtil nu i hvert fald.

Reply:
du er ikke nervøs for læsbarhed ift. sammensætning af list comprehension? nu har jeg ændret det til:

derived_parameters = [parameter for list_of_derived_parameters in [
            agent.shared_parameters[i].derived_parameters for
            i in keys(agent.shared_parameters)
        ] for parameter in list_of_derived_parameters]